### ISSUE TITLE
[Codespaces] Python Debugging for single Dockerfiles does not work in VS Codespaces

### ISSUE THREAD

**Issue by ucheNkadiCode** [microsoft/vscode-docker#2565]
After creating a working repo for a [Python application](https://github.com/microsoft/python-sample-vscode-django-tutorial/tree/tutorial) in VS Codespaces. 

When you run the command **Docker: Add files to workspace**, I selecting Python: Django with all of the defaults. This created a launch configuration named **Docker: Python - Django**. Unfortunately, upon F5 there seemed to be some sort of failure in Volume Mapping the debugger. Maybe there is an issue with the scaffolding of the launch configuration. 

Workaround: In order to debug I had to create a Python: Remote Attach configuration in `launch.json`. I then ran the command **Docker: Add Compose Files**, ran compose up on a container, then attached to to the running container. 

I would like to be able to debug my Python application in a container with only a Dockerfile.

**Comment by karolz-ms** [comment]
Thanks @ucheNkadiCode! Let's explore if we can avoid the volume mapping for Python by installing the debugger directly into the container.

@bwateratmsft feel free to steal this from me if necessary

**Comment by bwateratmsft** [comment]
We can, but it carries the risk of incompatibilities with the Python extension. Since we need volume mappings for .NET debugging to work, it's simpler if we just require the same for Python.

**Comment by bwateratmsft** [comment]
Additionally: either it would have to be in the Dockerfile (not pretty), or it would have to be done on the running container, which would add a small amount of time to the inner loop. Not a lot, debugpy is small, but some.

**Comment by bwateratmsft** [comment]
This is also covered by #2389, but I'll leave this issue open; we plan to copy in debugpy in the short term so we'll use this to track that.

**Comment by bwateratmsft** [comment]
We talked about this yesterday in a meeting, but we realized that volume mapping is probably not the only problem for Python. The debugging flow involves starting a debugpy server on the host machine, and starting a debugpy client within the container. That client is given information on how to connect back to the server. Generally this is done through the host gateway (Linux), or just simply 'localhost' (Mac, Win), but I doubt this will work in Codespaces' Docker-beside-Docker scenario.
